### ISSUE TITLE
Unexpected / Wrong (?) indentation if subtype has multiline arguments

### ISSUE THREAD

**Issue by elexx** [pinterest/ktlint#3333]
## Expected Behavior
The IndentationRule should allow for
```kotlin
 // This is already valid and accepted
class A : B({
    1
})

// This fails:
class X(
) : Y({
    1
})
```

## Observed Behavior (current `master` branch)
Currently valid and accepted
```kotlin
class A : B({
    1
})

class X(
) : Y({
        1
    })
```

## Background
Im currently implementing tests with [Kotest](https://kotest.io/) and [Spring](https://spring.io/), which look like this, when auto-formatted by ktlint. The first testclass got an odd indention, because it has a multiline argumentlist (the injected helloService), the second testclass looks good - no unexpected indentation.
```kotlin
class MyTestClass(
    helloService: IHelloService,
) : FunSpec({
        test("expect that sayHello return Hello") {
            helloService.sayHello() shouldBe "Hello"
        }
    })

class MyOtherTestClass : FunSpec({
    test("expect that Hello equals Hello") {
        "Hello" shouldBe "Hello"
    }
})
```

## Test
I wrote a unittest in [IndentationRuleTest.kt](https://github.com/ktlint/ktlint/blob/master/ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/IndentationRuleTest.kt) to verify it is not some config sideeffect in my project. This test currently fails:
```kotlin
    @Test
    fun `Given a value argument list with lambda in super type entry and multiline arguments sub type`() {
        val code =
            """
            class A(
            ) : B({
                1
            })

            open class B(f: () -> Int)
            """.trimIndent()
        indentationRuleAssertThat(code).hasNoLintViolations()
    }
```

**Comment by paul-dingemans** [comment]
Hi @elexx,

Tnx for creating the issue. This problem has been reported before in #2902. The formatting that you request is quite specific for Kotest. I do understand why you want to all your tests to be formatted similarly. However, in my opinion it is a code smell to have an empty argument list to be written as a multiline argument list.

As advised in https://github.com/ktlint/ktlint/issues/2902#issuecomment-2551733329:

> You can disable this rule only for test classes. You have two options:
>
> Add a .editorconfig in the root folder of the test classes
> Exclude the rule in the root .editorconfig. For example, see https://pinterest.github.io/ktlint/latest/faq/#how-do-i-disable-ktlint-for-generated-code

**Comment by elexx** [comment]
Thanks for advising and pointing me to #2902. Is there an option to only disable indentation for type definitions? Completly disabling it for all unittests is a bit too drastic.

I agree, a multiline empty argument list is codesmell! I might have gone too far with my minimal example unittest. I just wanted to boil it down as far as possible. The problem also exists with non-empty multiline argument lists - see my real world example with kotest. In my opinion the indentation should be the same, whether the inheritence is written is single-line or multiline. Currently this is the valid indentation:
```kotlin
class A(
    i: Int
) : B(
        1, // why twice indentation here?
    ) // why indentation here?
class A(i: Int) : B(
    1,
)
```

**Comment by paul-dingemans** [comment]
This has also been discussed in other issues before. Please see https://github.com/ktlint/ktlint/issues/2661#issuecomment-2119275841
